### PR TITLE
Droth 721 new speed limit for state roads

### PR DESCRIPTION
**Pull request by leirasf**

### CODE REVIEW

File: UI/src/asset-type-layer-specifications.js [hunk 1]
```diff
@@ -413,7 +423,24 @@
   ];
 
   root.experimentalLinearAssetSpecs = [
-  // In future this array could be use to include another experimental Linear
+    {
+      typeId: assetType.trSpeedLimits,
+      singleElementEventCategory: 'trSpeedLimit',
+      multiElementEventCategory: 'trSpeedLimits',
+      layerName: 'trSpeedLimits',
```
Comment (joanacoelho) [inline UI/src/asset-type-layer-specifications.js]: the suggested name should be: "speedLimitState"

**Comment by marquesfm** [comment]
Seems ok for me

File: UI/src/view/PointAssetForm.js [hunk 2]
```diff
@@ -38,60 +41,60 @@
...
+    var eventTarget = $(event.currentTarget);
+    selectedAsset.set({name: eventTarget.val()});
+  });
+
+  rootElement.find('.form-directional-traffic-sign textarea').on('keyup', function (event) {
+    var eventTarget = $(event.currentTarget);
+    selectedAsset.set({text: eventTarget.val()});
+  });
+
+  rootElement.find('.linear-asset.form textarea').on('keyup', function (event) {
```
Comment (joanacoelho) [inline UI/src/view/PointAssetForm.js]: Comment related with US 750 - Probably this element should have another name!

File: digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/DataFixture.scala [hunk 3]
```diff
@@ -980,14 +1060,27 @@ object DataFixture {
...
         " split_speedlimitchains | split_linear_asset_chains | dropped_assets_csv | dropped_manoeuvres_csv |" +
         " unfloat_linear_assets | expire_split_assets_without_mml | generate_values_for_lit_roads | get_addresses_to_masstransitstops_from_vvh |" +
         " prohibitions | hazmat_prohibitions | european_roads | adjust_digitization | repair | link_float_obstacle_assets |" +
         " generate_floating_obstacles | import_VVH_RoadLinks_by_municipalities | " +
         " check_unknown_speedlimits | set_transitStops_floating_reason | verify_roadLink_administrative_class_changed | set_TR_bus_stops_without_OTH_LiviId |" +
         " check_TR_bus_stops_without_OTH_LiviId | check_bus_stop_matching_between_OTH_TR | listing_bus_stops_with_side_code_conflict_with_roadLink_direction |" +
-        " fill_lane_amounts_in_missing_road_links | import_all_trafficVolume_from_TR_to_OTH | import_all_litRoad_from_TR_to_OTH")
+        " fill_lane_amounts_in_missing_road_links | import_all_trafficVolume_from_TR_to_OTH | import_all_litRoad_from_TR_to_OTH | import_all_roadWidth_from_TR_to_OTH |" +
+        " update_litRoad_from_TR_to_OTH")
```
Comment (joanacoelho) [inline digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/DataFixture.scala]: Missing `import_all_speedLimits_from_TR_to_OTH`

**Comment by joanacoelho** [comment]
Ok to me

File: UI/src/router.js [hunk 4]
```diff
@@ -34,14 +34,16 @@
...
         'work-list/speedLimit': 'speedLimitWorkList',
         'work-list/linkProperty': 'linkPropertyWorkList',
         'work-list/massTransitStop': 'massTransitStopWorkList',
         'work-list/pedestrianCrossings': 'pedestrianCrossingWorkList',
         'work-list/trafficLights': 'trafficLightWorkList',
         'work-list/obstacles': 'obstacleWorkList',
         'work-list/railwayCrossings': 'railwayCrossingWorkList',
-        'work-list/directionalTrafficSigns': 'directionalTrafficSignsWorkList'
+        'work-list/directionalTrafficSigns': 'directionalTrafficSignsWorkList',
+        'work-list/trafficSigns': 'trafficSigntWorkList'
```
Comment (oskarmalmstrom) [inline UI/src/router.js]: typo in  'trafficSigntWorkList' -> 'trafficSignsWorkList'

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TierekisteriImporterOperations.scala [hunk 5]
```diff
@@ -0,0 +1,569 @@
...
+          //Generate all speed limits of the left side of the road
+          generateOneSideSpeedLimits(roadNumber, RoadSide.Left, trAssets)
+        }
+    }
+  }
+
+  override protected def createLinearAsset(roadLink: VVHRoadlink, roadAddress: ViiteRoadAddress, section: AddressSection, measures: Measures, trAssetData: TierekisteriAssetData) = {
+    val speedLimit = getSpeedLimitValue(trAssetData)
+    if (measures.startMeasure != measures.endMeasure) {
+      val assetId = linearAssetService.dao.createLinearAsset(typeId, roadLink.linkId, false, getSideCode(roadAddress.sideCode, trAssetData.roadSide).value,
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TierekisteriImporterOperations.scala]: Check if the asset value is between 0 and 120

**Pull request merged**